lib/starttls.js

```diff
@@ -33,111 +15,40 @@ function starttlsClient(socket, options, callback) {
 }
 
 function starttls(socket, options, callback, isServer) {
-  var sslcontext;
-
   var opts = {};
+
   Object.keys(options).forEach(function(key) {
     opts[key] = options[key];
   });
   if (!opts.ciphers) {
     opts.ciphers = RECOMMENDED_CIPHERS;
   }
+  opts.isServer = isServer;
+  // NodeJS bug: opts are not passed to implicitly created context
+  // https://github.com/nodejs/node/issues/10538
```

Comment:
not a bug, its a missing feature - its never done that, I just incorrectly claimed it did in the docs (sorry). Even when I add the feature, it may or may not make its way all the way back to v4.x, so it would be better to keep doing things this way, it will work across all LTS versions.